tests/e2e/linalg_ext_ops/attention.mlir

```diff
@@ -27,6 +30,76 @@ func.func @attention1x3x4() {
+  %mask = util.unfoldable_constant dense<[[[true, false,  false],
+                                           [true, true,   false],
+                                           [true, true,   true]]]> : tensor<1x3x3xi1>
+  %scale = arith.constant 0.5 : f32
+  %1 = iree_linalg_ext.attention  {indexing_maps = [affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d3, d2)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d3, d4)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> ()>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d3)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d4)>]}
+                     ins(%query, %key, %value, %scale, %mask : tensor<1x3x4xf32>,
+        tensor<1x3x4xf32>, tensor<1x3x4xf32>, f32, tensor<1x3x3xi1>) outs(%init : tensor<1x3x4xf32>) {
+          ^bb0(%arg0: f32):
+          iree_linalg_ext.yield %arg0 : f32
+        } -> tensor<1x3x4xf32>
+  check.expect_almost_eq_const(
+      %1,
+      dense<[[[0.1000, 0.2000, 0.3000, 0.4000],
+              [0.3509, 0.4509, 0.5509, 0.6509],
+              [0.7011, 0.8011, 0.9011, 1.0011]]]> : tensor<1x3x4xf32>
+  ) : tensor<1x3x4xf32>
+  return
+}
+
+
+func.func @softcap_attention1x3x4() {
+  %init = tensor.empty() : tensor<1x3x4xf32>
+  %query = util.unfoldable_constant dense<[[[-0.6523,  0.7802, -0.6770, -0.9474],
+                                            [ 0.6822,  0.8594,  0.1035, -0.5115],
+                                            [-0.7860, -0.5562, -0.2750, -0.0973]]]> : tensor<1x3x4xf32>
+
+  %key = util.unfoldable_constant dense<[[[-0.3457,  0.5223, -0.0553,  0.5851],
+                                          [ 0.4057, -0.8352,  0.1931,  0.1678],
+                                          [ 0.1438,  0.7713, -0.6535,  0.9221]]]> : tensor<1x3x4xf32>
+  %value = util.unfoldable_constant dense<[[[-0.2679,  0.6395,  0.4953,  0.3751],
+                                            [ 0.8195, -0.5137,  0.2178,  0.9182],
+                                            [ 0.8644,  0.7615,  0.2934,  0.9721]]]> : tensor<1x3x4xf32>
+  %scale = arith.constant 1.0 : f32
+  %1 = iree_linalg_ext.attention  {indexing_maps = [affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d3, d2)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d3, d4)>,
+                     affine_map<(d0, d1, d2, d3, d4) -> ()>,
+                     affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d4)>]}
+                     ins(%query, %key, %value, %scale : tensor<1x3x4xf32>,
+        tensor<1x3x4xf32>, tensor<1x3x4xf32>, f32) outs(%init : tensor<1x3x4xf32>) {
+          ^bb0(%arg0: f32):
+          %tanh = math.tanh %arg0 : f32
+          iree_linalg_ext.yield %tanh : f32
+        } -> tensor<1x3x4xf32>
+  check.expect_almost_eq_const(
```

Comment:
Can you add examples where the region has an arguement to it? Like a division by constant maybe?

Reply:
I can't add more arguments but I can add constants.